### PR TITLE
Resolves #19 ESDC "[Plugins] Method Adder"

### LINKED ISSUE #19: [Plugins] Method Adder
Add ability for a plugin to add a method that will be exposed by the Bush instance.

### PR DESCRIPTION
**Pull request by mortimr**
Resolves #19 ESDC

### CODE REVIEW

File: ts_sources/sources/Bush.test.ts [hunk 1]
```diff
@@ -117,6 +117,45 @@ const interrupt_mdw: MiddlewareFunction<ReceiveDT, any> = (data: ReceiveDT, env:
...
+            this.done();
+        });
+        bush.expand('done', () => {
+            this.done();
+        });
+    }
+}
+
+// tslint:disable-next-line:max-classes-per-file
+class ExpandingPlugin implements IBushPlugin {
```
Comment (Insomniiak) [inline ts_sources/sources/Bush.test.ts]: ```// tslint:disable-next-line:max-classes-per-file ```
Hmm I think this is not very good

File: ts_sources/sources/Bush.test.ts [hunk 2]
```diff
@@ -117,6 +117,45 @@ const interrupt_mdw: MiddlewareFunction<ReceiveDT, any> = (data: ReceiveDT, env:
 
 // Test Plugins
 
+// tslint:disable-next-line:max-classes-per-file
```
Comment (Insomniiak) [inline ts_sources/sources/Bush.test.ts]: ```// tslint:disable-next-line:max-classes-per-file```
Hmm I think this is not very good

Comment (mortimr) [reply ts_sources/sources/Bush.test.ts]: This is a test file.

File: ts_sources/sources/Bush.test.ts (hunk 1 above)
Comment (mortimr) [reply ts_sources/sources/Bush.test.ts]: This is a test file.

**Pull request merged**